specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/preview/2023-01-01-preview/Workspaces.json

```diff
@@ -0,0 +1,1002 @@
+          },
+          {
+            "name": "location",
+            "in": "query",
+            "required": true,
+            "type": "string",
+            "description": "The secondary location for the workspace."
+          } 
+        ],
+        "responses": {
+          "202": {
+            "description": "Accepted",
+            "headers": {
+              "Location": {
+                "type": "string"
+              }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "location"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.OperationalInsights/workspaces/{workspaceName}/disableSecondaryLocation": {
+      "post": {
+        "tags": [
+          "Workspaces"
+        ],
+        "x-ms-examples": {
+          "WorkspacesDisableSecondaryLocation": {
+            "$ref": "./examples/WorkspacesDisableSecondaryLocation.json"
+          }
+        },
+        "operationId": "Workspaces_DisableSecondaryLocation",
+        "description": "Disables a secondary location for a workspace.",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/WorkspaceNameParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "202": {
+            "description": "Accepted",
+            "headers": {
+              "Location": {
+                "type": "string"
+              }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "location"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.OperationalInsights/locations/{location}/workspaces/{workspaceName}/failover": {
+      "post": {
```

Comment:
As discussed please raise the 2 workitems we spoke about that will be addressed before public preview

Reply:
Note to reviewer: Please refer to this PR in the private repo for more context - [https://github.com/Azure/azure-rest-api-specs-pr/pull/11332](https://github.com/Azure/azure-rest-api-specs-pr/pull/11332)

The ARM manifest cab be merged once the links to the work items have been provided.